src/core/qgsvectorfilewriter.cpp

```diff
@@ -284,12 +284,12 @@ void QgsVectorFileWriter::init( QString vectorFileName,
   {
     if ( metadataFound )
     {
-      QStringList allExts = metadata.ext.split( ' ', Qt::SkipEmptyParts );
+      QStringList allExts = metadata.glob.split( ' ', Qt::SkipEmptyParts );
       bool found = false;
       const auto constAllExts = allExts;
       for ( const QString &ext : constAllExts )
       {
-        if ( vectorFileName.endsWith( '.' + ext, Qt::CaseInsensitive ) )
+        if ( vectorFileName.endsWith( '.' + ext.mid( 2 ), Qt::CaseInsensitive ) )
```

Comment:
A code comment indicating what this 2 is, or a constexpr const int SYMBOLIC_NAME = 2 would be helpful